internal/common/baseclient/baseclient.go

```diff
@@ -69,5 +72,8 @@ func (c *BaseClient) StartOfTurnUpdate(gameState gamestate.ClientGameState) {
 // The gameState is served by the server.
 // OPTIONAL. Base should be able to handle it but feel free to implement your own.
 func (c *BaseClient) GameStateUpdate(gameState gamestate.ClientGameState) {
+	c.clientInfo = gameState.ClientInfo.Copy()
```

Comment:
`.Copy` not needed :) -- it was already passed in as a copy.

Can't the turn and season be acquired from `c.clientInfo.Turn` as well? I don't see the need of the two extra `uint` vars.

Do we need this for `StartOfTurnUpdate` as well?

Reply:
whoops didn't notice these comments! will do that now

Reply:
@lhl2617 i don't see that clientInfo has Turn and Season..?

Reply:
Whoops sorry! yeah it doesnt :) -- how about just defining a `clientGameState` varin the struct then? :)

Reply:
ah ok sure - shall i make a copy function in clientgamestate? i don't think i get when we need a copy func and when not ...

Reply:
Good question! 

Outside of this code, the short answer is "same as Java and Python". Some types are *primitive*, and they are always _passed by copy_ (such as ints, strings, floats). Other types are *reference types* and they are _passed by value of the reference_ (in golang: maps, slices, channels, functions). 

In Python we'd need to `deepcopy` the reference types to obtain a true copy, similar here.

As when the server calls `GameStateUpdate` the ClientGameState is already (deep-)copied, you don't need to call it anywhere in client side.

Reply:
The risk of not copying is that the reference passed to the client means that the client has access to change the reference types to some bogus value. So all copying should be done _before_ passing it to the client.

Reply:
ahhhhh, i see - thank you :) 
i'll make a new pr seeing as this already merged